notation/install.ts

```diff
@@ -0,0 +1,46 @@
+import * as path from 'path';
+import * as taskLib from 'azure-pipelines-task-lib/task';
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+
+import { getDownloadInfo, installFromURL } from './lib/install';
+
+const NOTATION_VERSION_FILE = 'notation_versions.json';
+
+export async function install(): Promise<void> {
+    // the notation download URL
+    let version: string
+    let downloadURL: string
+    let checksum: string
+
+    if (taskLib.getBoolInput('isCustomVersion', false)) {
+        // for custom version, download the notation from specified URL
+        version = "1.0.0-UNRELEASED"
+        downloadURL = taskLib.getInput('url', true) || '';
+        checksum = taskLib.getInput('checksum', false) || '';
+    } else {
+        const versionRange = taskLib.getInput('version', true);
+        if (!versionRange) {
+            throw new Error('Version is not specified');
+        }
+        const downloadInfo = getDownloadInfo(versionRange, NOTATION_VERSION_FILE);
+        version = downloadInfo.version;
+        downloadURL = downloadInfo.url;
+        checksum = downloadInfo.checksum;
+    }
+
+    // install notation binary
+    const extractPath = taskLib.getVariable('Agent.TempDirectory');
+    if (!extractPath) {
+        throw new Error('Agent.TempDirectory is not set');
```

Comment:
The `Agent.TempDirectory` is defined by pipeline, so the exception will not be triggered if it is running on Azure Pipelines. 
![image](https://github.com/AzureCR/notation-ado-task/assets/43160897/f5a756b0-776a-49b0-89ba-64a085f5b260)
If I removed the check, there will be an error because `taskLib.getVariable` return type is `string | undefined`. By adding the check, the value will be `string`